Documentation/project-docs/analyzers.md

```diff
@@ -0,0 +1,27 @@
+## Adding new analyzers to the build
+
+This repo relies on [.NET Compiler Platform analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019) to help validate the correctness, performance, and maintainability of the code.  Several existing analyzer packages are wired into the build, but it is easy to augment the utilized packages in order to experiment with additional analyzers.
+
+To add an analyzer package to the build:
+1. Select a package you want to employ, for example https://www.nuget.org/packages/SonarAnalyzer.CSharp/.  This analyzer package's name is `SonarAnalyzer.CSharp` and the latest version as of this edit is `7.15.0.8572`.
+2. Add a PackageReference entry to https://github.com/dotnet/corefx/blob/master/eng/Analyzers.props, e.g.
+```XML
+<PackageReference Include="SonarAnalyzer.CSharp" Version="7.15.0.8572" />
+```
+3. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  Rules can be disabled by adding entries to the https://github.com/dotnet/corefx/blob/master/src/CodeAnalysis.ruleset file, e.g.
+```XML
+  <Rules AnalyzerId="SonarAnalyzer.CSharp" RuleNamespace="SonarAnalyzer.CSharp">
+    <Rule Id="S1006" Action="None" />
+    <Rule Id="S1075" Action="None" />
+    ...
```

Comment:
Is there a way to get an export of the full ruleset with the summaries like our current ones have?  (Then the analyzer-adder can change them all to None manually)

Reply:
Not that I know of.  What I did was:
1. build and get all the warnings.
2. wrote a little post-processor that found all unique IDs and a description for each from the output and spit out the XML entries for each
3. copy/pasted those into the .ruleset

This meant that I was disabling all rules for which we currently had warnings, but leaving enabled rules where we were currently clean.

@sharwell?